src/components/sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect, useRef } from "react";
+import { Link } from "react-router-dom";
+import {
+  HistoryIcon,
+  HomeIcon,
+  LikeIcon,
+  PlaylistIcon,
+  WatchLaterIcon,
+} from "../../assets";
+import "./sidebar.css";
+import { useWindowResize } from "beautiful-react-hooks";
+import { useGlobalEvent, useDebouncedCallback } from "beautiful-react-hooks";
+
+const Sidebar = () => {
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+  const onWindowResize = useGlobalEvent("resize");
```

Comment:
Appreciation: woah I've seen useGlobalEvent(), will read about it.

Reply:
Yeah. You should definitely read about it!